### PR TITLE
Task#24 - Feat: Show product details and product's price list in card basket 

### LINKED ISSUE #90: User Story#1: Display Cart Summary Sidebar
> As a user, I want a button that, when clicked, triggers a slide-in sidebar from the right, occupying one-third of the screen. This sidebar should display a list of products that have been added to the cart.

- #91 

- [x] Design a prominent button labeled "Cart" and place it in a visible location on the screen.
- [x] Define the appearance of the slide-in sidebar, including its dimensions, color scheme, and transition animation.
- [x] Implement the functionality to display the sidebar when the "Cart" button is clicked.

- #93

- [x] Retrieve the list of products that have been added to the cart and display them in the sidebar.
- [x] Scrolls appear if the product is too much
- [x] Show UI product names, quantities, individual prices, and a total cost calculation.
- [x] Show a button so the user can Proceed to Checkout

- Task#3: Test Cart Summary Sidebar

- [x] Test the interaction of the "Cart" button and the slide-in sidebar.
- [x] Ensure the sidebar functions smoothly on various devices and screen sizes.
- [x] Verify that the cart contents are accurately displayed and updated.

### LINKED ISSUE #93: Task#24 - DevTask: Show product and product's price list in card basket
- #89 
  - #90  

## OVERVIEW
Products details: 
- Show images of products
- Show the name of the products
- Show icon to remove products from the cart
- Show prices of products
- Show UI quantity of products

Total price:
- Display tables according to UI design
- Show a button so the user can Proceed to Checkout

## ACCEPTANCE

- [x] Retrieve the list of products added to the cart and display them in the sidebar.
- [x]  Scrolls appear if the product is too much
- [x]  Show UI product names, quantities, individual prices, and a total cost calculation.
- [x]  Show a button so the user can Proceed to Checkout

## DEMO 
[Link](https://next-commerce-v2.vercel.app/matte-black-magic-mug)

Products details: 
image

Total price:
image

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #93 

## OVERVIEW

- Create product details in the cart
- Create a total price of products
- Pass data to display product list
- Update props for Shopping cart components

## DEMO

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/5109e555-db11-4e28-a326-68cf33234e37

### CODE REVIEW

File: practice-two/src/components/ShoppingCart.jsx [hunk 1]
```diff
@@ -45,17 +48,171 @@ const ShoppingCart = () => {
...
+            <div className="flex h-full w-full flex-col justify-between overflow-hidden p-1 pb-10">
+              <ul className="flex-grow overflow-auto py-4">
+                {products.map((product) => {
+                  const firstImage = Object.values(product.images)[0]; // Get the first image of the current product
+                  return (
+                    <li
+                      key={product.id}
+                      className="flex w-full flex-col border-b border-neutral-300 dark:border-neutral-700"
+                    >
+                      <div className="relative flex w-full flex-row justify-between px-1 py-4">
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ShoppingCart.jsx]: Split into component

File: practice-two/src/components/ShoppingCart.jsx [hunk 2]
```diff
@@ -45,17 +48,171 @@ const ShoppingCart = () => {
...
+                            </Button>
+                          </div>
+                        </div>
+                      </div>
+                    </li>
+                  );
+                })}
+              </ul>
+              {/* Total */}
+              <div className="py-4 text-sm text-neutral-500  dark:text-neutral-400">
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ShoppingCart.jsx]: Please create component for this

File: practice-two/src/components/ShoppingCart.jsx (hunk 1 above)
Comment (TheHienDuong) [reply practice-two/src/components/ShoppingCart.jsx]: Done

File: practice-two/src/components/ShoppingCart.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-two/src/components/ShoppingCart.jsx]: Done

File: practice-two/src/components/CartItem.jsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+// Product Image Component
+const ProductImage = ({ src, alt }) => (
+  <div className="relative h-16 w-16 cursor-pointer overflow-hidden rounded-md border border-neutral-300 bg-neutral-300 dark:border-neutral-700 dark:bg-neutral-900 dark:hover:bg-neutral-800">
+    <img src={src} alt={alt} width="64" height="64" className="h-full w-full object-cover" />
+  </div>
+);
+
+// Custom Typography Component
+const CustomTypography = ({
```
Comment (phattruong-agilityio) [inline practice-two/src/components/CartItem.jsx]: Do we need this?

File: practice-two/src/components/ShoppingCart.jsx [hunk 4]
```diff
@@ -45,17 +50,36 @@ const ShoppingCart = () => {
...
+            </div> */}
+            <div className="flex h-full w-full flex-col justify-between overflow-hidden p-1 pb-10">
+              <ul className="flex-grow overflow-auto py-4">
+                {products.map((product) => (
+                  <CartItem key={product.id} product={product} />
+                ))}
+              </ul>
+              <TotalSection />
+              <Link
+                href="/"
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ShoppingCart.jsx]: ??

**Review by phattruong-agilityio — APPROVED** [review]

File: practice-two/src/components/CartItem.jsx (hunk 3 above)
Comment (TheHienDuong) [reply practice-two/src/components/CartItem.jsx]: This is not necessary, Removed this component

File: practice-two/src/components/ShoppingCart.jsx (hunk 4 above)
Comment (TheHienDuong) [reply practice-two/src/components/ShoppingCart.jsx]: Updated into href="/Payment "

File: practice-two/src/components/CartItem.jsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      className="hover:text-accent-3 mx-[1px] h-4 w-4 text-white dark:text-black"
+    >
+      <path strokeLinecap="round" strokeLinejoin="round" d="M6 18L18 6M6 6l12 12" />
+    </svg>
+  </Button>
+);
+
+// Quantity Button Component
+const QuantityButton = ({ label = '', onClick = () => {}, children }) => (
+  <Button // github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/prop-types.mdon
```
Comment (phattruong-agilityio) [inline practice-two/src/components/CartItem.jsx]: ??

Comment (TheHienDuong) [reply practice-two/src/components/CartItem.jsx]: Removed

**Review by phattruong-agilityio — APPROVED** [review]

**Pull request merged**